test/WebJobs.Script.Tests/WorkerFunctionMetadataProviderTests.cs

```diff
@@ -140,5 +143,32 @@ public void ValidateBindings_OutputNameWithoutDirection_Throws()
 
             Assert.Equal($"{ScriptConstants.SystemReturnParameterBindingName} bindings must specify a direction of 'out'.", ex.Message);
         }
+
+        [Fact]
+        public async void ValidateFunctionMetadata_Logging()
+        {
+            var logger = new TestLogger<WorkerFunctionMetadataProvider>();
+            logger.ClearLogMessages();
+
+            var workerConfigs = TestHelpers.GetTestWorkerConfigs().ToImmutableArray();
+            workerConfigs.ToList().ForEach(config => config.Description.WorkerIndexing = "true");
+
+            var scriptApplicationHostOptions = new ScriptApplicationHostOptions();
+            var optionsMonitor = TestHelpers.CreateOptionsMonitor(scriptApplicationHostOptions);
+
+            var mockWebHostRpcWorkerChannelManager = new Mock<IWebHostRpcWorkerChannelManager>();
+            mockWebHostRpcWorkerChannelManager.Setup(m => m.GetChannels(It.IsAny<string>())).Returns(() => new Dictionary<string, TaskCompletionSource<IRpcWorkerChannel>>
+            {
+            });
+
+            var workerFunctionMetadataProvider = new WorkerFunctionMetadataProvider(optionsMonitor, logger, SystemEnvironment.Instance, mockWebHostRpcWorkerChannelManager.Object);
+            await workerFunctionMetadataProvider.GetFunctionMetadataAsync(workerConfigs, false);
+
+            var traces = logger.GetLogMessages();
+
+            // Assert that the logs contain the expected messages
+            Assert.Equal(2, traces.Count);
+            Assert.Equal("Reading functions metadata (Worker)", traces[1].FormattedMessage);
```

Comment:
Is there also a "x functions found (Custom)" assert you can add here for completeness?

Reply:
So "x functions found (Custom)" doesn't appear in the `WorkerFunctionMetadataProvider` but I added the other log statement that does appear.